backend/src/layer/logic/api_command/command_handler.rs

```diff
@@ -0,0 +1,211 @@
+
+pub struct CommandHandler<D, A, I>
+where
+    D: CommandDataAccess + Sync + Send,
+    A: AdminNotification + Sync + Send,
+    I: ImageHoster + Sync + Send,
+{
+    command_data: D,
+    admin_notification: A,
+    image_hoster: I,
+    auth: Authenticator,
+}
+
+impl<D, A, I> CommandHandler<D, A, I>
+where
+    D: CommandDataAccess + Sync + Send,
+    A: AdminNotification + Sync + Send,
+    I: ImageHoster + Sync + Send,
+{
+    /// A function that creates a new [`CommandHandler`]
+    ///
+    /// # Errors
+    /// Throws an error, if the api keys could not be gotten from [`command_data`]
+    pub async fn new(command_data: D, admin_notification: A, image_hoster: I) -> Result<Self> {
+        let keys: Vec<String> = command_data
+            .get_api_keys()
+            .await?
+            .into_iter()
+            .map(|x| x.key)
+            .collect();
+        Ok(Self {
+            command_data,
+            admin_notification,
+            image_hoster,
+            auth: Authenticator::new(keys),
+        })
+    }
+
+    fn will_be_hidden(date: Date, report_count: u32) -> bool {
+        Self::get_date_difference(date) < 30 && report_count > Self::get_report_barrier(date)
+    }
+
+    fn get_date_difference(date: Date) -> i64 {
+        let today = Local::now().date_naive();
+        let difference = today.signed_duration_since(date);
+        difference.num_days()
+    }
+
+    fn get_report_barrier(date: Date) -> u32 {
+        let t = Self::get_date_difference(date) as f64;
+        (REPORT_FACTOR * t * t).floor() as u32 + 5
+    }
+}
+
+#[async_trait]
+impl<D, A, I> Command for CommandHandler<D, A, I>
+where
+    D: CommandDataAccess + Sync + Send,
+    A: AdminNotification + Sync + Send,
+    I: ImageHoster + Sync + Send,
+{
+    /// Command to report an image. It als gets checked whether the image shall get hidden.
+    async fn report_image(
+        &self,
+        image_id: Uuid,
+        reason: ReportReason,
+        auth_info: AuthInfo,
+    ) -> Result<()> {
+        if let Some(auth_info) = auth_info {
+            let image_command_type = ImageCommandType::ReportImage(reason);
+            self.auth
+                .authn_image_command(&auth_info, image_id, image_command_type)?;
+            let info = self.command_data.get_image_info(image_id).await?;
+            if !info.approved {
+                self.command_data
+                    .add_report(image_id, auth_info.client_id, reason)
+                    .await?;
+                let image_got_hidden = Self::will_be_hidden(info.upload_date, info.report_count);
+                if image_got_hidden {
+                    self.command_data.hide_image(image_id).await?;
+                }
+                let report_info = ImageReportInfo {
+                    reason,
+                    image_id,
+                    image_got_hidden,
+                    image_link: info.image_url,
+                    report_count: info.report_count,
```

Comment:
Mind thath the current report is not part of the info gotten from the database.
Maybe there is a better way (better place) to add this report, as this also has to be considerd inside `Self::will_get_hidden`. Maybe just increment the counter on the info.

Reply:
as the additional report also may influence the image rank, maybe we should re-request the image info from the database modifications?
or we should make clear that these stats are the state before the report?